src/pytorch_lightning/loops/optimization/manual_loop.py

```diff
@@ -42,7 +42,7 @@ class ManualResult(OutputResult):
     def from_training_step_output(cls, training_step_output: Optional[STEP_OUTPUT]) -> "ManualResult":
         extra = {}
         if isinstance(training_step_output, dict):
-            extra = {k: v for k, v in training_step_output.items() if k != "hiddens"}
+            extra = training_step_output
```

Comment:
Maybe this should still create a copy to maintain the old behaviour

```suggestion
            extra = training_step_output.copy()
```